File: Java/SelectionSort.java
public class SelectionSort {

    public static void main(String agrs[]){

        int a[] = {6,3,2,1,5,4};
        int n = a.length;
        int temp;


        for(int i = 0; i < n - 1; i++){
            int min = i;
            for(int j = i + 1; j < n; j++){
                if(a[j] < a[min]){
                    min = j;
                }
               
            }
            temp = a[i];
            a[i] = a[min];
            a[min] = temp;
        }

        for(int i : a){
            System.out.print(i + " ");
        }






    }
    
}

File: Data Structures/SelectionSort.c
#include<stdio.h>
#include<stdlib.h>


void SelectionSort(int a[], int n){
    for(int i = 0; i < n; i++){
        int iMin = i;
        for(int j = 0; j < n; j++){
            if(a[iMin] < a[j]){
                iMin = j;
            }
            int temp = a[i];
            a[i] = a[iMin];
            a[iMin] = temp;
        }
    }
}


int main(){

    int a[] = {2, 7, 4, 1, 5, 3};
    SelectionSort(a, 6);

    for(int i = 0; i < 6; i++){
        printf(" %d", a[i]);
    }   
    printf("\n");
    return 0;
}
